ext/gd/gd.c

```diff
@@ -4288,9 +4292,10 @@ PHP_FUNCTION(imageresolution)
 		RETURN_TRUE;
 	}
 
-	array_init(return_value);
-	add_next_index_long(return_value, gdImageResolutionX(im));
-	add_next_index_long(return_value, gdImageResolutionY(im));
+	array_init_size(return_value, 2);
```

Comment:
There's `zend_new_pair` for an optimized version of returning a packed array of 2 elements